### PR TITLE
Flag --cull-admin-users added (default: true) for when --cull-users=true

### PR DESCRIPTION
**Pull request by anderoonies**
## PR Summary by Erik

`--cull-users=true` culled/deleted JupyterHub users from the JupyterHub database independently of admin status. This PR adds the `--cull-admin-users` flag to control if admin users should our shouldn't also be culled.

To use this opt-in feature, you would pass `--cull-users=true --cull-admin-users=false`.

### CODE REVIEW

File: jupyterhub_idle_culler/__init__.py [hunk 1]
```diff
@@ -475,6 +479,16 @@ def main():
             """
         ).strip(),
     )
+    define(
+        "cull_admin_users",
+        type=bool,
+        default=True,
+        help=dedent(
+            """
+            Whether admin users should be culled (only if --cull_users=true).
```
Comment (consideRatio) [inline jupyterhub_idle_culler/__init__.py:488]:
```suggestion
            Whether admin users should be culled (only if --cull-users=true).
```

**Review by consideRatio — APPROVED** [review]
Makes sense to me! I added two suggestions that I think are slight improvements but without actual change.

File: jupyterhub_idle_culler/__init__.py [hunk 2]
```diff
@@ -320,9 +321,12 @@ def handle_user(user):
             # which introduces the 'created' field which is never None
             inactive = age
 
+        user_is_admin = user["admin"]
+
         should_cull = (
             inactive is not None and inactive.total_seconds() >= inactive_limit
-        )
+        ) and not (user_is_admin and not cull_admin_users)
```
[the remark this answers is not in the document]
Comment (anderoonies) [reply jupyterhub_idle_culler/__init__.py:328]: yeah, that looks like DeMorgan's law to me 😅  make sense, mine is pretty unreadable

**Comment by anderoonies** [comment]
@consideRatio pushed up with those changes!

**Comment by consideRatio** [comment]
Thank you @anderoonies for your work on this :heart: :tada:!

**Pull request merged**